enabler/src/components/payment-methods/card/card.ts

```diff
@@ -1,13 +1,25 @@
-import { ComponentOptions, PaymentMethod } from '../../../payment-enabler/payment-enabler';
+import { ComponentOptions, PaymentComponent, PaymentComponentBuilder, PaymentMethod } from '../../../payment-enabler/payment-enabler';
 import buttonStyles from '../../../style/button.module.scss';
 import inputFieldStyles from '../../../style/inputField.module.scss';
 import styles from '../../../style/style.module.scss';
 import { BaseComponent, BaseOptions } from '../../base';
 import { addFormFieldsEventListeners, fieldIds, getCardBrand, getInput, validateAllFields } from './utils';
 
+export class CardBuilder implements PaymentComponentBuilder {
```

Comment:
In [adyen-integration](https://github.com/commercetools/connect-payment-integration-adyen/blob/main/enabler/src/components/payment-methods/card.ts#L20), it creates `BaseComponentBuilder` and it is inherited by `CardBuilder`. Is there any benefit you found to inherit directly from `PaymentComponentBuilder`?

Reply:
Thanks @leungkinghin-ct. I will check it again

Reply:
@leungkinghin-ct pls, help me understand the purpose of using a `BaseComponentBuilder` by inheritance which I presume will implement the `PaymentComponentBuilder` interface. Do we want to promote reusing some code?

Which component should we return in the `build`?

```typescript
export abstract class BaseComponentBuilder
  implements PaymentComponentBuilder
{
  public componentHasSubmit = true;

  protected baseOptions: BaseOptions;
  protected paymentMethod: PaymentMethod;

  constructor(paymentMethod: PaymentMethod, baseOptions: BaseOptions) {
    this.paymentMethod = paymentMethod;
    this.baseOptions = baseOptions;
  }

  build(config: ComponentOptions): PaymentComponent {
    // return ?
  }
}
```

Reply:
@abreu-ct I checked the code in adyen repo once again, `BaseComponentBuilder` is used due to supporting multiple payment methods in front-end. For the template repo here, we only use card as the single payment method. Therefore I think it is alright to inherit from `PaymentComponentBuilder`

Reply:
Thank you @leungkinghin-ct. If we later on introduce a new payment method we can revisit this.